specs/_features/maxeb_increase/capella.py

```diff
@@ -1627,6 +1652,83 @@ def process_pending_balance_deposits(state: BeaconState) -> None:
     state.pending_balance_deposits = state.pending_balance_deposits[next_pending_deposit_index:]
 
 
+def apply_pending_consolidation(state: BeaconState, pending_consolidation: PendingConsolidation) -> None:
+    current_epoch = get_current_epoch(state)
+
+    # Resolve the target of consolidation
+    target_index = resolve_consolidated_to(state, pending_consolidation.target_index)
+    target_validator = state.validators[target_index]
+    source_validator = state.validators[pending_consolidation.source_index]
+
+    # Apply consolidation
+    source_validator.consolidated_to = target_index
```

Comment:
Looks like this logic will accept two consolidations such that:
```
PendingConsolidation { source: 0, target: 1 },
PendingConsolidation { source: 1, target: 0 }
```
If that tuple is processed, `resolve_consolidated_to` will cause an infinite loop

Reply:
Say the validators are v1 and v2. After the first consolidation is processed, v1.consolidated_to = 1. When the second consolidation is processed, target_index = resolve_consolidated_to(state, 0) = v1.consolidated_to = 1, so v2.consolidated_to is just to 1 (itself). 

I don't think anything bad happens here, other than it being a useless operation. Still, we could add `assert(target_index != pending_consolidation.source_index)`. 

We could also prohibit consolidating to an already consolidated target altogether, with `assert(state.validators[pending_consolidation.target_index].consolidated_to == UNSET_CONSOLIDATED_TO)`. I don't see why someone would ever need to consolidated to a validator v which is already consolidated to v', instead of just consolidating to v' in the first place, so I would just prohibit this and simplify the cases we need to be concerned with

Reply:
It is reasonable to me to skip consolidation if it appears to be happening to the same address.

While aborting consolidation to an already consolidated target doesn’t sound right to me. Consider the following consolidations: A -> B, B -> C — this order would lead to A, B consolidated to C making C liable for A’s and B’s slashable offences. If the order of these consolidations is the opposite and consolidation A -> B aborted then we get C liable for B’s slashings and not for A’s ones. It doesn’t seem to be a big issue in a normal case, but in general it is detrimental to incentives balance between different forks which can become a problem in some edge cases.

Reply:
actually, as we’re finalizing the consolidation flow before applying it, a surface of edge cases where an imbalance can cause a problem should be negligible 🤔

Reply:
Applied the suggestion of aborting consolidation if the target appears to be consolidated already